addons/source-python/packages/source-python/translations/strings.py

```diff
@@ -250,11 +244,29 @@ def get_string(self, language=None, **tokens):
             # Possibly raise an error silently here
             return ''
 
-        # Update the stored tokens with the given ones
-        self.tokens.update(tokens)
+        # Expose all TranslationStrings instances in self.tokens
+        exposed_tokens = {}
+        for token_name, token in self.tokens.items():
+            if isinstance(token, TranslationStrings):
+
+                # Pass additional tokens - these will be used to format
+                # the string
+                token = token.get_string(language, **tokens)
+
+            exposed_tokens[token_name] = token
+
+        # Expose all TranslationsStrings instances in **tokens
```

Comment:
I think we should create a helper method for this duplicate. Maybe something like this:
```python
        # Code...
        
        # Expose all TranslationStrings instances in self.tokens
        exposed_tokens = {}
        
        self._update_exposed_tokens(exposed_tokens, language, self.tokens, **tokens)
        self._update_exposed_tokens(exposed_tokens, language, tokens)
        
        return self[language].format(**exposed_tokens)
    
    @staticmethod
    def _update_exposed_tokens(exposed_tokens, language, tokens, **kwargs):
        for token_name, token in tokens.items():
            if isinstance(token, TranslationStrings):
                # Pass additional tokens - these will be used to format
                # the string
                token = token.get_string(language, **kwargs)

            exposed_tokens[token_name] = token
```